packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -15,9 +19,10 @@ import { useRef, useCallback, useLayoutEffect } from '@wordpress/element';
  * callback will be called with `null` and the new ref callback will be called
  * with the same node.
  *
- * @param {Array<RefObject|RefCallback>} refs The refs to be merged.
+ * @template {import('react').Ref<any>} T
+ * @param {Array<T>} refs The refs to be merged.
  *
- * @return {RefCallback} The merged ref callback.
+ * @return {import('react').RefCallback<TypeFromRef<T>>} The merged ref callback.
```

Comment:
We used to do `import('@wordpress/element')` and now we do `import('react')` for `RefCallback`. I'm curious if there's a good reason for that.

Reply:
`@wordpress/element` doesn't actually export all of the types needed, so importing directly from `react` is better/more consistent. More discussion here: https://github.com/WordPress/gutenberg/pull/31603#discussion_r628315729